test/Microsoft.AspNetCore.SignalR.Tests/DefaultHubLifetimeManagerTests.cs

```diff
@@ -128,7 +128,38 @@ public class DefaultHubLifetimeManagerTests
         }
 
         [Fact]
-        public async Task InvokeConnectionAsyncOnNonExistentConnectionNoops()
+        public async Task SendAllAsyncSendsToAllConnectionsEvenWhenSomeFailToSend()
+        {
+            using (var client = new TestClient())
+            using (var client2 = new TestClient())
+            {
+                var manager = new DefaultHubLifetimeManager<MyHub>();
+
+                var connectionMock = HubConnectionContextUtils.CreateMock(client.Connection);
+                var connectionMock2 = HubConnectionContextUtils.CreateMock(client2.Connection);
+
+                var tcs = new TaskCompletionSource<object>();
+                var tcs2 = new TaskCompletionSource<object>();
+                // Force an exception when writing to connection
+                connectionMock.Setup(m => m.WriteAsync(It.IsAny<HubMessage>())).Callback(() => tcs.TrySetResult(null)).Throws(new Exception("Message"));
+                connectionMock2.Setup(m => m.WriteAsync(It.IsAny<HubMessage>())).Callback(() => tcs2.TrySetResult(null)).Throws(new Exception("Message"));
+                var connection = connectionMock.Object;
+                var connection2 = connectionMock2.Object;
+
+                await manager.OnConnectedAsync(connection).OrTimeout();
+                await manager.OnConnectedAsync(connection2).OrTimeout();
+
+                var exception = await Assert.ThrowsAsync<Exception>(() => manager.SendAllAsync("Hello", new object[] { "World" }).OrTimeout());
```

Comment:
Not a huge fan of this, you get an exception from your hub method if a connection failed to send to

Reply:
It's always been that way (at least in ASP.NET Core). Not sure what SignalR 2.x did. I think you do want a chance to know about this in the application, so I do like that it's sent. Though of course, in the Redis scenario, you don't get any notification if another server fails to send to a client. Maybe logging and dumping is sufficient here.